lib/smart_answer/calculators/state_pension_age_calculator.rb

```diff
@@ -10,28 +10,30 @@ def initialize(answers)
       @gender = answers[:gender] ? answers[:gender].to_sym : nil
     end
 
-    def state_pension_date
-      StatePensionDateQuery.state_pension_date(dob, gender)
+    def state_pension_date(*args)
+      args << gender unless non_binary?
+      StatePensionDateQuery.state_pension_date(dob, args.last)
     end
 
     def can_apply?
       Time.zone.today >= earliest_application_date
     end
 
-    def pension_on_feb_29?
-      state_pension_date.month == 2 && state_pension_date.day == 29
+    def pension_on_feb_29?(*args)
+      state_pension_date(args.last).month == 2 && state_pension_date(args.last).day == 29
     end
 
-    def state_pension_age
-      if birthday_on_feb_29? && !pension_on_feb_29?
+    def state_pension_age(*args)
```

Comment:
I see you've added a splat argument here to account for sometimes having no arguments, and sometimes having one (either `:male` or `:female`). I think a default argument would do the job more cleanly:

```suggestion
    def state_pension_age(gender_for_calculation = gender)
```

Then you would replace all the `args.last` calls with `gender_for_calculation`.

Reply:
Thanks, Chris! I tried that and for some reason, it didn't work, probably something else was wrong. I've changed it now!